C/common/config_category.cpp

```diff
@@ -168,13 +171,44 @@ ConfigCategory::~ConfigCategory()
  */
 ConfigCategory& ConfigCategory::operator=(ConfigCategory const& rhs)
 {
+	m_name = string(rhs.m_name);
+	m_description = string(rhs.m_description);
+
 	for (auto it = rhs.m_items.cbegin(); it != rhs.m_items.cend(); it++)
 	{
 		m_items.push_back(new CategoryItem(**it));
 	}
 	return *this;
 }
 
+/**
+ * Set the m_value from m_default for each item
+ */
+void ConfigCategory::setItemsValueFromDefault()
+{
+	for (auto it = m_items.cbegin(); it != m_items.cend(); it++)
+	{
+		(*it)->m_value = string((*it)->m_default);
```

Comment:
Do we need the extra string() or can we just let operator= in the string class do it?

Reply:
Removed extra string() calls